src/chains/ethereum/ethereum/src/forking/fork.ts

```diff
@@ -295,13 +290,16 @@ export class Fork {
         let hardfork;
         // hardforks are iterated from earliest to latest
         for (const hf of common.hardforks()) {
-          if (hf.block === null) continue;
-          if (blockNumber >= BigInt(hf.block)) {
+          if (hf.timestamp) {
+            const hfTimestamp = BigInt(hf.timestamp);
+            if (block.timestamp >= hfTimestamp) {
+              hardfork = hf.name;
+            }
```

Comment:
Can we do `else { break; }` here? Since we rely on the `hardforks()` list to be in the order already?

Reply:
yep, likewise in the blocknumber case